### ISSUE TITLE
Missing implicit any error in destructuring with computed property

### ISSUE THREAD

**Issue by ajafff** [Microsoft/TypeScript#23334]
**TypeScript Version:**  2.9.0-dev.20180411

**Search Terms:** 

**Code**

```ts
// --noImplicitAny
let named = 'foo';
let {[named]: prop} = {prop: 'foo'};
```

**Expected behavior:**
`Element implicitly has an 'any' type because type '{ prop: string; }' has no index signature.`

**Actual behavior:**
`prop` is `any`, but there is no implicit any error

**Playground Link:** https://agentcooper.github.io/typescript-play/#code/DYUwLgBAdghgtiAJhAvBA5AMwPbfQbgChRIBvAbVgUQF0AuCABwCdtGBfVCUlthrXOnb4gA

**Related Issues:**
https://github.com/Microsoft/TypeScript/pull/23297#issuecomment-380004267

**Comment by weswigham** [comment]
If `named` is `const`, we report the error as anticipated; so the issue here is simply that a computed property name of type `string` used to destructure something without an index signature to `any` should be an `implicitAnyIndexError`.

**Comment by mattmccutchen** [comment]
Duplicate of #20725 (or vice versa).

**Comment by mhegazy** [comment]
@mattmccutchen i think #20725 is a different issue, and should be fixed in 2.9.
